Binary files qt3d/.git/index and qt3d_orig/.git/index differ
diff -ur qt3d/src/threed/scene/qglscenenode.cpp qt3d_orig/src/threed/scene/qglscenenode.cpp
--- qt3d/src/threed/scene/qglscenenode.cpp	2014-06-04 16:54:11.000000000 -0600
+++ qt3d_orig/src/threed/scene/qglscenenode.cpp	2014-06-04 16:55:58.000000000 -0600
@@ -354,6 +354,7 @@
 {
     Q_D(QGLSceneNode);
     d->geometry = geometry;
+    invalidateBoundingBox();
     emit updated();
 }
 
diff -ur qt3d/src/threed/viewing/qglview.cpp qt3d_orig/src/threed/viewing/qglview.cpp
--- qt3d/src/threed/viewing/qglview.cpp	2014-06-04 16:54:11.000000000 -0600
+++ qt3d_orig/src/threed/viewing/qglview.cpp	2014-06-04 16:54:04.000000000 -0600
@@ -52,6 +52,7 @@
 #include <QtCore/qtimer.h>
 #include <QtCore/qdatetime.h>
 #include <QtCore/qdebug.h>
+#include <QtCore/qmath.h>
 
 QT_BEGIN_NAMESPACE
 
@@ -1379,16 +1380,12 @@
     } else {
         // enable this to get wheel navigation that actually zooms by moving the
         // camera back, as opposed to making the angle of view wider.        
-        QVector3D viewVector= camera()->eye() - camera()->center();
-        qreal zoomMag = viewVector.length();
-        qreal zoomIncrement = -float(delta) / 100.0f;
-        if (!qFuzzyIsNull(zoomIncrement))
+        QVector3D viewVector = camera()->eye() - camera()->center();
+        qreal zoomIncrement  = -float(delta) / 120.0f;
+        qreal zoomMag = qPow(1.15f, zoomIncrement);
+        if (!qFuzzyIsNull(zoomMag))
         {
-            zoomMag += zoomIncrement;
-            if (zoomMag < 1.0f)
-                zoomMag = 1.0f;
-
-            QRay3D viewLine(camera()->center(), viewVector.normalized());
+            QRay3D viewLine(camera()->center(), viewVector);
             camera()->setEye(viewLine.point(zoomMag));
         }
     }
